HomeKitBridge/accessories/garageDoorOpener.php

```diff
@@ -0,0 +1,192 @@
+    {
+        $this->switchDevice($this->data['CurrentDoorState'], $value);
+    }
+
+    public function setCharacteristicTargetDoorState($value)
+    {
+        $this->switchDevice($this->data['TargetDoorState'], $value);
+    }
+
+    public function setCharacteristicObstructionDetected($value)
+    {
+        $this->switchDevice($this->data['ObstructionDetected'], $value);
+    }
+
+    protected function switchDevice($variableID, $value)
+    {
+        $targetVariable = IPS_GetVariable($variableID);
+
+        if ($targetVariable['VariableCustomAction'] != '') {
+            $profileAction = $targetVariable['VariableCustomAction'];
+        } else {
+            $profileAction = $targetVariable['VariableAction'];
+        }
+
+        if ($profileAction < 10000) {
+            echo 'No action was defined!';
+
+            return;
+        }
+
+        if ($targetVariable['VariableType'] == 0 /* Boolean */) {
+            $value = boolval($value);
+        } elseif ($targetVariable['VariableType'] == 1 /* Integer */) {
+            $value = intval($value);
+        } elseif ($targetVariable['VariableType'] == 2 /* Float */) {
+            $value = floatval($value);
+        } else {
+            echo 'Strings are not supported';
+
+            return;
+        }
+
+        if (IPS_InstanceExists($profileAction)) {
+            IPS_RunScriptText('IPS_RequestAction(' . var_export($profileAction, true) . ', ' . var_export(IPS_GetObject($variableID)['ObjectIdent'], true) . ', ' . var_export($value, true) . ');');
+        } elseif (IPS_ScriptExists($profileAction)) {
+            IPS_RunScriptEx($profileAction, ['VARIABLE' => $variableID, 'VALUE' => $value, 'SENDER' => 'WebFront']);
+        }
+    }
+}
+
+class HAPAccessoryConfigurationGarageDoorOpener
+{
+    public static function getPosition()
+    {
+        return 80;
+    }
+
+    public static function getCaption()
+    {
+        return 'Garage Door Opener';
+    }
+
+    public static function getColumns()
+    {
+        return [
+            [
+                'label' => 'CurrentDoorState',
+                'name'  => 'CurrentDoorState',
+                'width' => '100px',
+                'add'   => 0,
+                'edit'  => [
+                    'type' => 'SelectVariable'
+                ],
+            ],
+            [
+                'label' => 'TargetDoorState',
+                'name'  => 'TargetDoorState',
+                'width' => '100px',
+                'add'   => 0,
+                'edit'  => [
+                    'type' => 'SelectVariable'
+                ],
+            ],
+            [
+                'label' => 'ObstructionDetected',
+                'name'  => 'ObstructionDetected',
+                'width' => '100px',
+                'add'   => 0,
+                'edit'  => [
+                    'type' => 'SelectVariable'
+                ],
+            ]
+
+        ];
+    }
+
+    public static function getStatus($data)
```

Comment:
Ich fand es eig. ganz gut, so kann man direkt alle sehen, die falsch sind.
Allerdings wie du sagtest, die Spalte ist zu klein.
Aber wenn man das ganze auseinander zieht, könnte man ja für jede Variable ein extra Status Feld in der Zeile anlegen?!